### ISSUE TITLE
Implement backpropagation

### ISSUE THREAD

**Issue by kamil271e** [kamil271e/ffnn-parallel-computing#7]
- [x] Define a loss function and calculate errors
- [x] Implement derivatives of activation functions
- [x] Adjust the weights
